### PR TITLE
BREAKING(assert,testing): remove `formatter` option from `assertEquals()`

### LINKED ISSUE #4891: to-do: investigate the viability of `formatter` option for `assertEquals()`
We need to decide whether this option is to be kept before achieving stabilization. See [documentation](https://jsr.io/@std/assert/doc/~/assertEquals).

One clear point against it is that it's a fourth argument for `assertEquals()`, which goes against the Deno Style Guide of having two arguments and a third option argument, max (see https://docs.deno.com/runtime/manual/references/contributing/style_guide#exported-functions-max-2-args-put-the-rest-into-an-options-object). If we are to keep it, we should merge `msg` and `formatter` into a third options argument.

### PR DESCRIPTION
**Pull request by iuioiua**
It seems this option is used in testing `@std/testing/snapshot`. I'm not sure how we should proceed, @kt3k.

Closes #4891

### CODE REVIEW

**Comment by kt3k** [comment]
I think https://github.com/denoland/deno_std/pull/4930 unblocks this PR

File: testing/asserts.ts [hunk 1]
```diff
@@ -92,18 +92,15 @@ export function assertArrayIncludes<T>(
...
  * @deprecated This will be removed in 1.0.0. Import from
  * {@link https://jsr.io/@std/assert | @std/assert} instead.
  */
 export function assertEquals<T>(
   actual: T,
   expected: T,
   msg?: string,
-  options: { formatter?: (value: unknown) => string } = {},
 ) {
-  asserts.assertEquals<T>(actual, expected, msg, options);
```
Comment (kt3k) [inline testing/asserts.ts:106]: How about leaving this unchanged because this API is already deprecated and removed at 1.0.0-rc.1

Comment (iuioiua) [reply testing/asserts.ts:106]: I think we should keep the APIs between `@std/assert` and `@std/testing/asserts` synchronized, as it makes any migration seamless. Also, keeping this unchanged would mean I'd have to re-implement `assertEquals()` here in its previous state.

Comment (kt3k) [reply testing/asserts.ts:106]: Ah ok. It depends on `@std/assert`.. I missed it.

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**